src/cheatcodes.rs

```diff
@@ -47,21 +56,76 @@ abigen!(
         function deal(address who, uint256 newBalance)
         function etch(address who, bytes calldata code)
         function setNonce(address account, uint64 nonce)
+        function getNonce(address account)
         function warp(uint256 timestamp)
     ]"#
 );
 
-impl<F: NodeCtx, S: WriteStorage, H: HistoryMode> DynTracer<S, H> for CheatcodeTracer<F> {
+impl<F: NodeCtx, S: WriteStorage, H: HistoryMode> DynTracer<S, SimpleMemory<H>>
+    for CheatcodeTracer<F>
+{
     fn before_execution(
+        &mut self,
+        _state: VmLocalStateData<'_>,
+        _data: BeforeExecutionData,
+        _memory: &SimpleMemory<H>,
+        _storage: StoragePtr<S>,
+    ) {
+        // if self.returndata.is_some() {
+        // if let Opcode::UMA(a) = _data.opcode.variant.opcode {
+        //     if let UMAOpcode::HeapRead = a {
+        //         dbg!(_data);
+        //     }
+        // }
+        // }
+    }
+
```

Comment:
```suggestion
```

Reply:
Fixed in 8c46e906342d378104c7b3573fab7826891b8531